src/plugins/dashboard/public/application/lib/dashboard_control_group.ts

```diff
@@ -118,7 +118,7 @@ export const syncDashboardControlGroup = async ({
   subscriptions.add(
     dashboardContainer
       .getInput$()
-      .pipe(distinctUntilKeyChanged('controlGroupInput'))
+      .pipe(debounceTime(10), distinctUntilKeyChanged('controlGroupInput'))
```

Comment:
Sorry - completely unrelated to this PR, but the addition of `debounceTime` made me think of something here 😆 

Is there a reason we don't have a debounce for control selections? For example, in `options_list_embeddable.tsx`, we could have something like this: 

```typescript
this.subscriptions.add(
  this.getInput$()
    .pipe(
      debounceTime(100),
      distinctUntilChanged((a, b) => isEqual(a.selectedOptions, b.selectedOptions))
    )
    .subscribe(async ({ selectedOptions: newSelectedOptions }) => { ... }
```

This would slow down the number of output filters if, for example, a user selects an option by accident and then quickly unselects it - I tested the above change locally and it seemed to work great, but just wanted to see if there was a reason we're not already doing this 😄

Reply:
Oh oh! It's happening in the container - that makes a lot of sense! It wasn't obvious to me that a debounce was happening **at all** - probably because, as you said, the debounce happening in the container is so short that there is virtually no delay between updates. Like I alluded to in the original comment, even a quick double click would trigger an update, which doesn't seem ideal - it's possible my fingers are just slow though 😆

I can definitely see an argument for making it a *tiny* bit longer by default - and it would be awesome if users could set this! Because for small dashboards the debounce makes little to no difference so you might as well make it small, but it's pretty annoying to make an accidental click in something like the flights dashboard because you have to wait for *two* updates and it's quite slow.

Reply:
That does make sense! As a quick change, we might do a 'bugfix' for 8.2 that makes the delay a little longer. I originally made it this short because I found that for small dashboards, it would feel a little laggy if you clicked an option and it waited 100ms for it to apply. So there's definitely some balance we can search for there.

Then for 8.3 we can add a configuration option to set the `debounceTime`, so authors can figure out what works best for their dashboards!